tools/ui-components/src/button/button.tsx

```diff
@@ -1,35 +1,66 @@
-import React, { useMemo } from 'react';
+import React, { useCallback, useMemo } from 'react';
 import { ButtonProps, ButtonSize, ButtonVariant } from './types';
 
-const defaultClassNames = ['cursor-pointer', 'inline-block', 'border-3'];
+const defaultClassNames = [
+  'relative',
+  'cursor-pointer',
+  'inline-block',
+  'border-3',
+  'active:before:w-full',
+  'active:before:h-full',
+  'active:before:absolute',
+  'active:before:inset-0',
+  'active:before:border-3',
+  'active:before:border-transparent',
+  'active:before:bg-gray-900',
+  'active:before:opacity-20',
+  'aria-disabled:cursor-not-allowed',
+  'aria-disabled:opacity-50'
+];
 
 const computeClassNames = ({
   size,
-  variant
+  variant,
+  disabled,
+  block
 }: {
   size: ButtonSize;
   variant: ButtonVariant;
+  disabled?: boolean;
+  block?: boolean;
 }) => {
   const classNames = [...defaultClassNames];
 
+  if (block) {
+    classNames.push('block', 'w-full');
+  }
+
   // TODO: support 'link' variant
   switch (variant) {
     case 'danger':
       classNames.push(
         'border-default-foreground-danger',
         'bg-default-background-danger',
         'text-default-foreground-danger',
-        'hover:bg-default-background-danger-hover',
-        'hover:text-default-foreground-danger-hover'
+        ...(disabled
+          ? []
+          : [
+              'hover:bg-default-background-danger-hover',
+              'hover:text-default-foreground-danger-hover'
+            ])
```

Comment:
This would be `&:hover:not([aria-disabled="true"])` if we were to write in CSS. Unfortunately Tailwind doesn't allow a `not` selector so I had to add a condition here.